apis/login.js

```diff
@@ -0,0 +1,239 @@
+*                   landmark:
+*                     type: string
+*                age:
+*                 type: integer
+*                mobile:
+*                 type: string
+*                gender:
+*                 type: string
+*                roles:
+*                  type: array
+*                  items:
+*                     type: integer
+*                email:
+*                 type: string
+*                password:
+*                 type: string
+*                avatar:
+*                 type: string
+*                managerId:
+*                 type: string
+*                createdBy:
+*                 type: string
+*                updatedBy:
+*                 type: string
+*                createdAt:
+*                 type: string
+*                 format: date
+*                updatedAt:
+*                 type: string
+*                 format: date
+*                isActive:
+*                 type: boolean
+*                isDeleted:
+*                 type: boolean
+*       400:
+*         description: Bad request
+*/
+
+const registerUser = async (req, res) => {
+  const validatePassword = (password) => {
+    const regex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
+    return (regex.test(password));
+  }
+
+  if (!req.body.name.trim() || !req.body.address.trim() || !req.body.mobile.trim() || !req.body.gender.trim() || req.body.roles.length == 0 || !req.body.email.trim() || !req.body.password.trim() || req.body.age < 1) {
+    return res.status(400).send("name,address,age,mobile,gender,role,email and password are required fields");
+  }
+
+  if (req.body.age < 0 || req.body.age > 60) {
+    return res.status(400).send("age must be within 60");
+  }
+
+  if (!validator.isEmail(req.body.email)) {
+    return res.status(400).send("please enter valid email");
+  }
+ 
+  if (!validatePassword(req.body.password)) {
+    return res.status(400).send("Password must contain atleast one lower,one upper,one special character,one digit,no blank spaces and length must be between 8-20 characters");
+  }
+
+  const hashedPassword = await bcrypt.hash(req.body.password, 10);
+  const add = JSON.parse(req.body.address)
+
+  let employee = new userModel({
+    name: req.body.name,
+    address: {
+      flat_details: add.flat_details,
+      landmark: add.landmark,
+      area: add.area 
+    },
+    age: req.body.age,
+    mobile: req.body.mobile,
+    gender: req.body.gender,
+    roles: req.body.roles.split(','),
+    email: req.body.email,
+    password: hashedPassword,
+    managerId: req.body.managerId,
+    createdBy: req.body.createdBy,
+    updatedBy: req.body.updatedBy,
+    avatar: req.file.path
+  })
+
+  let data = await userService.createUser(employee);
+  res.send(data);
+}
+
+const authenticateToken = (req, res, next) => {
+  const token = req.headers.authorization?.split(' ')[1];
+  if (!token) {
+    return res.status(401).json({ error: 'Unauthorized: Token missing' });
```

Comment:
Any sort of response should come from resourcing files and should not be a hard coded string